webapp/channels/src/components/drafts/draft_row.tsx

```diff
@@ -156,59 +172,229 @@ function DraftRow({
         dispatch(removeDraft(key, channelId, rootId));
     }, [dispatch, channelId, rootId]);
 
+    const afterSubmit = useCallback((response: SubmitPostReturnType) => {
+        // if draft was being scheduled, delete the draft after it's been scheduled
+        if (isBeingScheduled.current && response.created && !response.error) {
+            handleOnDelete();
+            isBeingScheduled.current = false;
+        }
+
+        // if scheduled posts was being sent, delete the scheduled post after it's been sent
+        if (isScheduledPostBeingSent.current && response.created && !response.error) {
+            dispatch(deleteScheduledPost((item as ScheduledPost).id, connectionId));
+            isScheduledPostBeingSent.current = false;
+        }
+    }, [connectionId, dispatch, handleOnDelete, item]);
+
+    // TODO LOL verify the types and handled it better
```

Comment:
Probably this is not needed anymore?

Reply:
Its needed as we have a backlog ticket somewhere to handle it.